annotations/templates/annotations/settings.html

```diff
@@ -1,71 +1,131 @@
+              {% csrf_token %}
+              {{ form.errors }}
+              {{ form.non_field_errors }}
+              <div class="form-group">
+              <label for="fullname">Full Name</label>
+              <input type="text" class="form-control" id="id_fullname" name="full_name" value="{{ full_name }}">
+              </div>
+              <div class="form-group">
+              <label for="email">Email Address</label>
+              <input type="email" class="form-control" id="id_email" name="email" value="{{ email }}">
+              </div>
+              <div class="form-group">
+              <label for="affiliation">Affiliation</label>
+              <input type="text" class="form-control" id="id_affiliation" name="affiliation" value="{{ affiliation }}">
+              </div>
+              <div class="form-group">
+              <label for="location">Location</label>
+              <input type="text" class="form-control" id="id_location" name="location" value="{{ location }}">
+              </div>
+              <div class="form-group">
+              <label for="link">Link</label>
+              <input type="text" class="form-control" id="id_link" name="link" value="{{ link }}">
+              </div>
+              
+              <input type="hidden" id="defaultImage" name="imagefile" value="{{ preview }}"/>
+              
+              <span class="pull-left">
+              <input type="submit" class="btn btn-primary" value="Submit" />
+              </span>
+ </form>
+
+ <br><br><br>
+  <div class="form-group">
+    <div class="panel-footer clearfix">
+              <label for="imagefile">Please upload an image</label>
+              <input type="file" id="file_input"/>
+              <p id="status">Please select a file</p>
+              <img id="preview"  src="{{ preview }}"/>
+             </div></div>
+
+
+
+
+              </div>
+              </div>
+          </div>
+                
+
+             
+    
 </div>
+<script type="text/javascript">
+   /*
+      Function called when file input updated. If there is a file selected, then
+      start upload procedure by asking for a signed request from the app.
+   */
+   function init_upload(){
+       var files = document.getElementById("file_input").files;
+       var file = files[0];
+       if(file == null){
+           alert("No file selected.");
+           return;
+       }
+       get_signed_request(file);
+   }
+   
+   
+   /*
+      Bind listeners when the page loads.
+   */
+   (function() {
+       document.getElementById("file_input").onchange = init_upload;
+   })();
+   
+   
+   /*
+       Function to get the temporary signed request from the Python app.
+       If request successful, continue to upload the file using this signed
+       request.
+   */
+   function get_signed_request(file){
+       var xhr = new XMLHttpRequest();
+       xhr.open("GET", "/sign_s3?file_name="+file.name+"&file_type="+file.type);
```

Comment:
Once you give the ``sign_s3`` route a name in ``urls.py`` (see line comment in that file), you should use reverse URL resolution here, e.g.:

```python
{% url "sign_s3" %}
```